File: lambdas/algorithms/searching/java-searching/src/main/java/LinearSearch.java
public class LinearSearch extends BaseSearch {

    @Override
    protected int search(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            iterations++; // each loop iteration
            comparisons++; // nums[i] == target comparison
            if (nums[i] == target) {
                return i;
            }
        }
        return -1;
    }
}

File: lambdas/algorithms/searching/python-searching/src/algorithms/linear_search.py
from src.base_search import BaseSearch
from typing import List

class LinearSearch(BaseSearch):
    def search(self, nums: List[int], target: int) -> int:
        for i in range(len(nums)):
            self.iterations += 1
            self.comparisons += 1
            
            if nums[i] == target:
                return i
                
        return -1